src/core/jit/context.rs

```diff
@@ -84,11 +84,11 @@ impl<'a> ResolverContextLike for Context<'a, ConstValue, ConstValue> {
     }
 
     fn is_query(&self) -> bool {
-        self.is_query
+        self.plan.is_query()
     }
 
-    fn add_error(&self, _error: ServerError) {
-        todo!()
+    fn add_error(&self, error: ServerError) {
+        self.plan.add_error(error.into())
```

Comment:
Plans should ideally remain immutable and reusable between queries. `Context` is a good place to keep mutable content instead.

Reply:
we could've reviewed this before merging https://github.com/tailcallhq/tailcall/pull/2518 and now this pr contains both changes.

better revert this merge and fix it separately